connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -387,4 +402,50 @@ boolean isHeartbeatTopic(String topic) {
     String formatRemoteTopic(String topic) {
         return replicationPolicy.formatRemoteTopic(sourceAndTarget.source(), topic);
     }
+
+    // Writing the translated offsets to the __consumer_offsets topic in the destination cluster,
+    // as long as no consumers in that group are connected to the destination cluster
+    private void syncCommitOffset() throws InterruptedException, ExecutionException {
+        MirrorClient mirrorClient = new MirrorClient(config.targetConsumerConfig());
+        Map<String, Map<TopicPartition, OffsetAndMetadata>> allOffsets =
+            mirrorClient.remoteConsumerOffsets(config.sourceClusterAlias(), config.adminTimeout());
+
+        // track the consumer that are not actively consuming from the replicated topic with same consumer group id
+        Set<String> idleConsumers = new HashSet<>();
+        DescribeConsumerGroupsResult describedGroups = targetAdminClient.describeConsumerGroups(allOffsets.keySet());
+        for (Map.Entry<String, KafkaFuture<ConsumerGroupDescription>> group : describedGroups.describedGroups().entrySet()) {
```

Comment:
We should also filter with the configured GroupFilter s.t. only whitelisted groups are sync'd this way. As written, anything with a checkpoint is sync'd, which is problematic when e.g. a user blacklists a group: we should stop sync'ing the group even tho a previous checkpoint may exist.

Reply:
I think once we move the "consumer offset sync" to `MirrorCheckpointTask`, the group in the blacklist will be filtered out by the existing logics in `MirrorCheckpointTask`